shm/inc/delegate.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <iostream>
+
+class Delegate {
+public:
+    virtual void payCrew(size_t money) = 0;
+    virtual ~Delegate() {}
+};
```

Comment:
Delegate should not be standalone class. You can create a lot of delegates and then you have conflict with names.